src/Neo/SmartContract/Native/CryptoLib.cs

```diff
@@ -63,6 +64,45 @@ public static byte[] Murmur32(byte[] data, uint seed)
             return murmur.ComputeHash(data);
         }
 
+
+        /// <summary>
+        /// Computes the script hash from specified public key byte array.
+        /// </summary>
+        /// <param name="pubKey">The public key to be used.</param>
+        /// <param name="curve">The curve to be used by the ECDSA algorithm.</param>
+        /// <returns>The computed public key script hash. Zero in case of error.</returns>
+        [ContractMethod(CpuFee = 1 << 15)]
+        public static UInt160 PubKeyToScriptHash(byte[] pubKey, NamedCurve curve)
+        {
+            try
+            {
+                return Contract.CreateSignatureRedeemScript(ECPoint.FromBytes(pubKey, curves[curve])).ToScriptHash();
+            }
+            catch (ArgumentException)
+            {
+                return UInt160.Zero;
+            }
+        }
+
+        /// <summary>
+        /// Computes the address from specified script hash.
+        /// </summary>
+        /// <param name="scriptHash">The script hash to be used.</param>
+        /// <param name="version">The version of the address.</param>
+        /// <returns>The computed address of script hash. Empty in case of error.</returns>
+        [ContractMethod(CpuFee = 1 << 14)]
+        public static string ScriptHashToAddr(UInt160 scriptHash, byte version)
```

Comment:
I'd pair this with `AddrToScriptHash`, although this can be a part of some non-standard library as well (like https://pkg.go.dev/github.com/nspcc-dev/neo-go/pkg/interop@v0.0.0-20230130121057-bd26b9db8307/lib/address).

Reply:
This is a good one, will try to add it to the compiler